Fix QSPIF Bus mode mask and quad enable write SR size

offirko opened the pull request:
Fix Bus mode mask, and fix status register write size in some quad enable modes

Fix: https://github.com/ARMmbed/mbed-os/issues/10165
Fix: https://github.com/ARMmbed/mbed-os/issues/10163

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

Commit:
Fix Bus mode mask and status register write size in
some modes in quad enable

0xc0170 commented on components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp:
```diff
@@ -874,6 +875,7 @@ int QSPIFBlockDevice::_sfdp_set_quad_enabled(uint8_t *basic_param_table_ptr)
             status_reg_setup[1] = 0x2; // Bit 1 of status Reg 2
             _read_register_inst = 0x35;
             sr_read_size = 1;
+	    sr_write_size = 2;
```
seems to be misaligned (astyle should catch these)

offirko replied:
sorry fixing

0xc0170 commented on components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp:
```diff
@@ -1032,7 +1034,7 @@ int QSPIFBlockDevice::_sfdp_detect_best_bus_read_mode(uint8_t *basic_param_table
         is_qpi_mode = false;
         examined_byte = basic_param_table_ptr[QSPIF_BASIC_PARAM_TABLE_FAST_READ_SUPPORT_BYTE];
-        if (examined_byte & 0x40) {
+        if (examined_byte & 0x20) {
```
how is this mode fix related to sr_write_size - they are both in the same commit - should they be in separate commits ?

offirko replied:
Its two small changes, both are related to quad enabling process... though they in different areas. I can separate them if required.

0xc0170 replied:
Could be fine, what about commit message (issues have details for these fixes to be tter understand them) ?

Commit:
Fixed Bus Mode bit mask to select best mode.
When setting Quad Enable, either SR1, SR2 or CR setup is required.
Either way register size is up to 2 bytes.

cmonr commented on components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp:
```diff
@@ -1045,7 +1047,7 @@ int QSPIFBlockDevice::_sfdp_detect_best_bus_read_mode(uint8_t *basic_param_table
         }
 
-        if (examined_byte & 0x80) {
+        if (examined_byte & 0x40) {
```
It bothers me that these are magic numbers instead of definitions somewhere, but I'm not going to block the PR on them.

It would be nice if they could be un-magic-fied at some point.

offirko replied:
@cmonr - support for each mode is defiend as a certain bit of a certain byte. I did use a macro for the Byte location. I'll use a definition for the bits as well at later delivery.
![image](https://user-images.githubusercontent.com/36957743/54741446-2aca8000-4bc7-11e9-9219-36e93a0d0cb0.png)

cy-vaira reviewed (commented):
> @VairaRamasamy - can you please test the fix on CY8CPROTO_062_4343W

@offirko I've tested this on CY8CPROTO_062_4343W and it works! Thanks for the fix.

yossi2le reviewed (approved):
I would consider adding a return in the default section of the case. 
Other than that, it looks good to me.

Commit:
If QER is undefined: do nothing but log warning and continue

Commit:
Update QSPIFBlockDevice.cpp
